[CI][DSA][Blackwell] oss:rel shard1 sparse attention backward Tensor-likes mismatch

## Summary

`oss:rel: [Blackwell, shard1]` is failing on Blackwell with `TEST_FAILURE` in DSA sparse attention backward. 

## CI job

Jobs:
- Job: `oss:rel: [Blackwell, shard1]` (#368076777, lane=parent, ref=develop)
- Job URL: [368076777](https://nv/cudnn_frontend/-/jobs/368076777)
- Parent FE pipeline: [58766219](https://nv/cudnn_frontend/-/pipelines/58766219)
- Anchor pipeline: 
  - Last successful pipeline [58113302](https://nv/cudnn_frontend/-/pipelines/58113302) 
  - First failing pipeline [58282151](https://nv/cudnn_frontend/-/pipelines/58282151) 

## Version window

- cuDNN frontend: [82b97bac](https://nv/cudnn_frontend/-/commit/82b97bac4f2251460838116eb47a0c909f88614d) → [b641fef0](https://nv/cudnn_frontend/-/commit/b641fef0ab931a1c9e8140d0ef9881223a5e5832)
- cuDNN backend: `9.25.0.11-cb2f1a05-58087465` → `9.25.0.11-cb2f1a05-5808746` (branch: `dev`)
- Container: `(cudnn/cudnn_frontend:cudnn_13.3.0)`
- CUDA: `13.3`
- cuDNN: `9.25.0.15-1c5ca8e8-58515561`
- GPU / SM: `NVIDIA Graphics Device` / `SM 10.0`

## Failing test

- First failing test: `test/python/fe_api/dsa/test_DSA_sparse_attention_backward.py::test_DSA_sparse_attention_backward_wrapper[False-512-576-512-32-acc_dtype0-dtype0]`
- Exit code: `1`

## Error snippet

```text
FAILED test/python/fe_api/dsa/test_DSA_sparse_attention_backward.py::test_DSA_sparse_attention_backward_wrapper[False-512-576-512-32-acc_dtype0-dtype0] - AssertionError: Tensor-likes are not close!
Mismatched elements: 15 / 32 (46.9%)
Greatest absolute difference: 0.8858855962753296 at index (5,) (up to 0.05 allowed)
Greatest relative difference: 1.0 at index (0,) (up to 0.05 allowed)
FAILED test/python/fe_api/dsa/test_DSA_sparse_attention_backward.py::test_DSA_sparse_attention_backward_wrapper[True-512-576-512-32-acc_dtype0-dtype0] - AssertionError: Tensor-likes are not close!
Mismatched elements: 97741 / 18874368 (0.5%)
```

## Bisect result:
- Target repository: cudnn_frontend
- Target commit: dc953fa4
- Target title: Support SM90 DSA qh16 indexer forward and fix qh32 sparse backward (#388)

## Root cause analysis:
- Commit dc953fa41 introduced/enabled the DSA sparse attention backward qh32 MLA shape (head_dim=576, head_dim_v=512, num_heads=32) in the L0 parameterization. This exact shape is the first failing test in the trace.
- The same commit rewrote the SM90 sparse_attention_backward scheduling/LSE+dPsum loading path in python/cudnn/deepseek_sparse_attention/sparse_attention_backward/dsa_bwd_sm90.py. The kernel now maps work tiles as one CTA per query with qhead tile indexing, then loads per-head LSE/dPsum via _load_f32_head_to_smem. For the qh32 MLA shape, the produced dq/dkv/d_sink do not match the PyTorch reference.
- Since both the passing anchor and first failing job use cuDNN 9.25.0.11-cb2f1a05, this is attributed to the cudnn_frontend FE change rather than a cuDNN backend version change.

## Reproduction
```text
pytest -q test/python/fe_api/dsa/test_DSA_sparse_attention_backward.py -k 'test_DSA_sparse_attention_backward_wrapper and 576 and 512 and 32'
```

--- a/python/cudnn/deepseek_sparse_attention/sparse_attention_backward/dsa_bwd_sm100.py
+++ b/python/cudnn/deepseek_sparse_attention/sparse_attention_backward/dsa_bwd_sm100.py
@@ -581,25 +581,24 @@ class FlashAttentionDSABackwardSm100:
             stream=stream,
         )
 
-        if cutlass.const_expr(self.same_hdim_kv):
-            dSink_grid = (
-                cute.ceil_div(problem_shape[0], self.dSink_block_q),
-                problem_shape[3][0],
-                problem_shape[3][1],
-            )
-            self.sum_dSink(
-                sum_OdO,
-                scaled_LSE,
-                mAttnSink,
-                mdSink,
-                problem_shape,
-            ).launch(
-                grid=dSink_grid,
-                block=[self.dSink_num_threads, 1, 1],
-                cluster=[1, 1, 1],
-                stream=stream,
-                min_blocks_per_mp=1,
-            )
+        dSink_grid = (
+            cute.ceil_div(problem_shape[0], self.dSink_block_q),
+            problem_shape[3][0],
+            problem_shape[3][1],
+        )
+        self.sum_dSink(
+            sum_OdO,
+            scaled_LSE,
+            mAttnSink,
+            mdSink,
+            problem_shape,
+        ).launch(
+            grid=dSink_grid,
+            block=[self.dSink_num_threads, 1, 1],
+            cluster=[1, 1, 1],
+            stream=stream,
+            min_blocks_per_mp=1,
+        )
 
     @cute.kernel
     def convert(
@@ -683,21 +682,20 @@ class FlashAttentionDSABackwardSm100:
                     lse_bhq = lse[bidy, (idx_q + offset, bidz)]
                     sum_OdO_bhq = sum_OdO_scale * acc
 
-                    if cutlass.const_expr(self.same_hdim_kv):
-                        attn_sink_bh = attn_sink[bidy, (0, bidz)]
+                    # FlashMLA LSE excludes the sink for every head-dim
+                    # specialization, so always fold it into the denominator.
+                    attn_sink_bh = attn_sink[bidy, (0, bidz)]
 
-                        log2_e = -lse_scale
-                        lse_log2 = lse_bhq * log2_e
-                        sink_log2 = attn_sink_bh * log2_e
-                        lse_max_log2 = cute.arch.fmax(lse_log2, sink_log2)
-                        sum_exp2 = Float32(cute.math.exp2(lse_log2 - lse_max_log2) + cute.math.exp2(sink_log2 - lse_max_log2))
-                        lse_with_sink_log2 = lse_max_log2 + cute.math.log2(sum_exp2)
-                        scaled_lse_bhq = -lse_with_sink_log2
+                    log2_e = -lse_scale
+                    lse_log2 = lse_bhq * log2_e
+                    sink_log2 = attn_sink_bh * log2_e
+                    lse_max_log2 = cute.arch.fmax(lse_log2, sink_log2)
+                    sum_exp2 = Float32(cute.math.exp2(lse_log2 - lse_max_log2) + cute.math.exp2(sink_log2 - lse_max_log2))
+                    lse_with_sink_log2 = lse_max_log2 + cute.math.log2(sum_exp2)
+                    scaled_lse_bhq = -lse_with_sink_log2
 
-                        if lse_bhq == Float32(float("inf")):
-                            scaled_lse_bhq = Float32(float("-inf"))
-                    else:
-                        scaled_lse_bhq = lse_scale * lse_bhq
+                    if lse_bhq == Float32(float("inf")):
+                        scaled_lse_bhq = Float32(float("-inf"))
 
                     sum_OdO[bidy, (idx_q + offset, bidz)] = sum_OdO_bhq
                     scaled_lse[bidy, (idx_q + offset, bidz)] = scaled_lse_bhq

--- a/test/python/fe_api/dsa/test_DSA_sparse_attention_backward.py
+++ b/test/python/fe_api/dsa/test_DSA_sparse_attention_backward.py
@@ -137,6 +137,80 @@ def test_DSA_sparse_attention_backward_wrapper(
 
 
 @pytest.mark.L0
+@pytest.mark.parametrize("has_topk_length", [False, True], ids=["full-topk", "lengths"])
+@torch_fork_set_rng(seed=418)
+def test_DSA_sparse_attention_backward_sm100_576_includes_sink_in_normalization(has_topk_length):
+    """The 576/512 specialization must include sink mass in every gradient."""
+    try:
+        from cudnn import DSA
+        from cuda.bindings import driver as cuda
+    except ImportError:
+        pytest.skip("Environment not supported: cudnn[cutedsl] not installed")
+
+    major, minor = torch.cuda.get_device_capability()
+    if major * 10 + minor < 100:
+        pytest.skip("sink-normalization regression test targets the SM100 kernel")
+
+    device = torch.device("cuda")
+    s_q = 5
+    s_kv = topk = 512
+    num_heads, head_dim = 32, 576
+    softmax_scale = 1.0 / math.sqrt(head_dim)
+
+    q = torch.randn(s_q, num_heads, head_dim, dtype=torch.bfloat16, device=device)
+    kv = torch.randn(s_kv, head_dim, dtype=torch.bfloat16, device=device)
+    # Keep the sink probability significant even for the full-top-k row so
+    # both compiled variants fail clearly if they normalize by KV-only LSE.
+    attn_sink = torch.full((num_heads,), math.log(topk), dtype=torch.float32, device=device)
+    topk_idxs = torch.arange(topk, dtype=torch.int32, device=device).expand(s_q, -1).contiguous()
+    lengths = torch.tensor([1, 63, 64, 65, 512], dtype=torch.int32, device=device)
+    topk_length = lengths if has_topk_length else None
+
+    out, lse = ref_sparse_attention_forward(
+        q,
+        kv,
+        attn_sink,
+        topk_idxs,
+        topk_length=topk_length,
+        softmax_scale=softmax_scale,
+    )
+    # O.dO is positive for every head, so a zero d_sink cannot pass by
+    # landing inside the absolute tolerance.
+    dout = out.clone()
+    stream = cuda.CUstream(torch.cuda.current_stream().cuda_stream)
+
+    result = DSA.sparse_attention_backward_wrapper(
+        q,
+        kv,
+        out,
+        dout,
+        lse,
+        attn_sink,
+        topk_idxs,
+        softmax_scale=softmax_scale,
+        topk_length=topk_length,
+        stream=stream,
+    )
+
+    check_ref_dsa_sparse_attention_backward(
+        q,
+        kv,
+        attn_sink,
+        topk_idxs,
+        out,
+        dout,
+        lse,
+        result["dq"],
+        result["dkv"],
+        result["d_sink"],
+        softmax_scale=softmax_scale,
+        topk_length=topk_length,
+        atol=5e-2,
+        rtol=5e-2,
+    )
+
+
+@pytest.mark.L0
 @torch_fork_set_rng(seed=385)
 def test_DSA_sparse_attention_backward_qh32_uses_per_query_topk_without_padding(monkeypatch):
     try:

